import { API } from "../axiosConfig";

const getAll = async () => {
    try {
        const res = await API.get("/api/check");

        return res.data;
    } catch (error) {
        if (error.response) {
            throw new Error(error.response.data.message || "Erro na resposta da API");
        } else if (error.request) {
            throw new Error("Sem resposta do servidor");
        } else {
            throw new Error(error.message);
        }
    };
};

const getById = async (id) => {
    try {
        const res = await API.get(`/api/check/${id}`);

        return res.data;
    } catch (error) {
        if (error.response) {
            throw new Error(error.response.data.message || "Erro na resposta da API");
        } else if (error.request) {
            throw new Error("Sem resposta do servidor");
        } else {
            throw new Error(error.message);
        }
    };
};

const getByStatus = async (status) => {
    try {
        const res = await API.get(`/api/check/status/${status}`);

        return res.data;
    } catch (error) {
        if (error.response) {
            throw new Error(error.response.data.message || "Erro na resposta da API");
        } else if (error.request) {
            throw new Error("Sem resposta do servidor");
        } else {
            throw new Error(error.message);
        }
    };
};

const create = async (data) => {
    try {
        const res = await API.post("/api/check", data);

        return res.data;
    } catch (error) {
        if (error.response) {
            throw new Error(error.response.data.message || "Erro na resposta da API");
        } else if (error.request) {
            throw new Error("Sem resposta do servidor");
        } else {
            throw new Error(error.message);
        }
    };
};

const createClosed = async (data) => {
    try {
        const res = await API.post("/api/check/closed", data);

        return res.data;
    } catch (error) {
        if (error.response) {
            throw new Error(error.response.data.message || "Erro na resposta da API");
        } else if (error.request) {
            throw new Error("Sem resposta do servidor");
        } else {
            throw new Error(error.message);
        }
    };
};

const updateById = async (id, data) => {
    try {
        const res = await API.put(`/api/check/${id}`, data);

        return res.data;
    } catch (error) {
        if (error.response) {
            throw new Error(error.response.data.message || "Erro na resposta da API");
        } else if (error.request) {
            throw new Error("Sem resposta do servidor");
        } else {
            throw new Error(error.message);
        }
    };
};

const closeCheck = async (pay_form, check_id) => {
    try {
        const res = await API.put(`/api/check/close/${check_id}`, { pay_form });

        return res.data;
    } catch (error) {
        if (error.response) {
            throw new Error(error.response.data.message || "Erro na resposta da API");
        } else if (error.request) {
            throw new Error("Sem resposta do servidor");
        } else {
            throw new Error(error.message);
        }
    };
};

const insetNotifyId = async (check_id, notify_id) => {
    try {
        const res = await API.put(`/api/check/insert_notify_id/${check_id}`, { notify_id });

        return res.data;
    } catch (error) {
        if (error.response) {
            throw new Error(error.response.data.message || "Erro na resposta da API");
        } else if (error.request) {
            throw new Error("Sem resposta do servidor");
        } else {
            throw new Error(error.message);
        }
    };
};

const deleteById = async (id) => {
    try {
        const res = await API.delete(`/api/check/${id}`);

        return res.data;
    } catch (error) {
        if (error.response) {
            throw new Error(error.response.data.message || "Erro na resposta da API");
        } else if (error.request) {
            throw new Error("Sem resposta do servidor");
        } else {
            throw new Error(error.message);
        }
    };
};

const deleteAll = async () => {
    try {
        const res = await API.delete(`/api/check/delete/delete_all`);

        return res.data;
    } catch (error) {
        if (error.response) {
            throw new Error(error.response.data.message || "Erro na resposta da API");
        } else if (error.request) {
            throw new Error("Sem resposta do servidor");
        } else {
            throw new Error(error.message);
        }
    };
};

export const CheckService = {
    create,
    createClosed,

    getAll,
    getById,
    getByStatus,

    updateById,
    closeCheck,
    insetNotifyId,

    deleteAll,
    deleteById,
};